pkg/scalers/external_scaler.go

```diff
@@ -273,19 +298,32 @@ var connectionPoolMutex sync.Mutex
 
 // getClientForConnectionPool returns a grpcClient and a done() Func. The done() function must be called once the client is no longer
 // in use to clean up the shared grpc.ClientConn
-func getClientForConnectionPool(metadata externalScalerMetadata) (pb.ExternalScalerClient, error) {
+func getClientForConnectionPool(metadata externalScalerMetadata, logger logr.Logger) (pb.ExternalScalerClient, error) {
 	connectionPoolMutex.Lock()
 	defer connectionPoolMutex.Unlock()
 
 	buildGRPCConnection := func(metadata externalScalerMetadata) (*grpc.ClientConn, error) {
+		// FIXME: DEPRECATED to be removed in v2.13
```

Comment:
Could you please open an issue to track this deprecation and link it here in the comment?

Reply:
Yes, thanks for the pointer, I have created a new issue for this https://github.com/kedacore/keda/issues/4549